internal/controller/nonadminbackup_controller.go

```diff
@@ -97,11 +144,279 @@ func (r *NonAdminBackupReconciler) Reconcile(ctx context.Context, req ctrl.Reque
+			return false, err
+		}
+		logger.V(1).Info("VeleroBackup deletion initiated", nameString, veleroBackup.Name)
+	} else {
+		logger.V(1).Info("VeleroBackup already deleted")
+	}
+
+	deleteBackupRequest, err := function.GetVeleroDeleteBackupRequestByLabel(ctx, r.Client, r.OADPNamespace, veleroBackupNameUUID)
+	if err != nil {
+		// Log error if multiple DeleteBackupRequest objects are found
+		logger.Error(err, findSingleVDBRError, uuidString, veleroBackupNameUUID)
+		return false, err
+	}
+	if deleteBackupRequest != nil {
+		if err = r.Delete(ctx, deleteBackupRequest); err != nil {
+			logger.Error(err, "Failed to delete VeleroDeleteBackupRequest", nameString, deleteBackupRequest.Name)
+			return false, err
+		}
+		logger.V(1).Info("VeleroDeleteBackupRequest deletion initiated", nameString, deleteBackupRequest.Name)
+	}
+	return false, nil // Continue so initNabDeletion can initialize deletion of an NonAdminBackup object
+}
+
+// removeNabFinalizerUponVeleroBackupDeletion ensures the associated VeleroBackup object is deleted
+// and removes the finalizer from the NonAdminBackup (NAB) object to complete its cleanup process.
+//
+// Parameters:
+//   - ctx: Context for managing request lifetime.
+//   - logger: Logger instance for logging messages.
+//   - nab: Pointer to the NonAdminBackup object undergoing cleanup.
+//
+// This function first checks if the `DeleteBackup` field in the NAB spec is set to true or if
+// the NAB has been marked for deletion by Kubernetes. If either condition is met, it verifies
+// the existence of an associated VeleroBackup object by consulting the UUID stored in the NAB’s status.
+// If the VeleroBackup is found, the function waits for it to be deleted before proceeding further.
+// After confirming VeleroBackup deletion, it removes the finalizer from the NAB, allowing Kubernetes
+// to complete the garbage collection process for the NAB object itself.
+//
+// Returns:
+//   - A boolean indicating whether to requeue the reconcile loop (true if waiting for VeleroBackup deletion).
+//   - An error if any update operation or deletion check fails.
+func (r *NonAdminBackupReconciler) removeNabFinalizerUponVeleroBackupDeletion(ctx context.Context, logger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (bool, error) {
+	// Check if DeleteBackup is set or NAB was deleted
+	if !nab.ObjectMeta.DeletionTimestamp.IsZero() {
+		if !nab.Spec.ForceDeleteBackup && nab.Status.VeleroBackup != nil && nab.Status.VeleroBackup.NameUUID != constant.EmptyString {
+			veleroBackupNameUUID := nab.Status.VeleroBackup.NameUUID
+
+			veleroBackup, err := function.GetVeleroBackupByLabel(ctx, r.Client, r.OADPNamespace, veleroBackupNameUUID)
+			if err != nil {
+				// Case in which more then one VeleroBackup is found with the same label UUID
+				// TODO: Should we delete all of the objects with such UUID ?
+				logger.Error(err, findSingleVBError, uuidString, veleroBackupNameUUID)
+				return false, err
+			}
+
+			if veleroBackup != nil {
```

Comment:
Hmm, this might not be sufficient, right ? We might also need to check up on DeleteBackupRequest associated with the VB ? The status of DBR might give more details to us and then based on that we can go further in the deletion workflow, WDYT ?

Reply:
Note this is different then https://github.com/migtools/oadp-non-admin/pull/108#discussion_r1830585938 because in this path we are certain DBR was removed. In the https://github.com/migtools/oadp-non-admin/pull/108#discussion_r1830585938 we do want to clean this without checking why that failed (force).

Reply:
agree with Michal here, I think current check is enough